### PR TITLE
feat: plugin descriptor json scheme generator

### PR DESCRIPTION
**Pull request by Cdm2883**

### CODE REVIEW

File: codegen/build.gradle.kts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 version = "1.0.0"
 
 dependencies {
+    implementation(project(":server"))
```
Comment (smartcmd) [inline codegen/build.gradle.kts:4]: codegen shouldn't depends on server

**Review by smartcmd — CHANGES_REQUESTED** [review]

**Comment by Cdm2883** [comment]
呜呜呜

**Review by Cdm2883 — COMMENTED** [review]
呜呜呜

**Comment by Cdm2883** [comment]
@smartcmd 呜呜呜

**Comment by YoumiHaLH** [comment]
> @smartcmd 呜呜呜

呜呜呜呜呜

**Comment by Cdm2883** [comment]
> > @smartcmd 呜呜呜
> 
> 呜呜呜呜呜

呜哇哇哇

**Comment by Cdm2883** [comment]
https://gist.github.com/Cdm2883/79e6e6ed371c9346c5d4914fea6a687c

**Pull request merged**